core/src/main/java/com/graphhopper/routing/profiles/RoadClass.java

```diff
@@ -1,55 +1,55 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
 package com.graphhopper.routing.profiles;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
+import com.graphhopper.util.Helper;
 
-public class RoadClass implements IndexBased {
-    private static final List<RoadClass> values = create("_default",
-            "motorway", "motorway_link", "motorroad",
-            "trunk", "trunk_link",
-            "primary", "primary_link",
-            "secondary", "secondary_link",
-            "tertiary", "tertiary_link",
-            "residential", "unclassified",
-            "service", "road", "track", "forestry", "steps", "cycleway", "path", "living_street");
+/**
+ * This class defines a IndexBased but is type safe.
+ */
+public enum RoadClass {
+    OTHER("other"), MOTORWAY("motorway"), MOTORROAD("motorroad"),
+    TRUNK("trunk"), PRIMARY("primary"), SECONDARY("secondary"),
+    TERTIARY("tertiary"), RESIDENTIAL("residential"), UNCLASSIFIED("unclassified"),
+    SERVICE("service"), ROAD("road"), TRACK("track"),
+    FORESTRY("forestry"), STEPS("steps"), CYCLEWAY("cycleway"),
```

Comment:
Why do we need FORESTRY here in RoadClass?

BTW: Would it make sense to add briddleway here?

Reply:
You are right. Forestry is not a highway (we would use track with access=forestry). And surely we can add briddleway.